tasks/deploy.py

```diff
@@ -0,0 +1,55 @@
+from . import config
+import os
+import invoke
+import getpass
+import subprocess
+import shlex
+
+def deploy(
+          ctx, 
+          scenario_name, 
+          install_agent=None, 
+          agent_version=None):
+    flags = {}
+    flags["scenario"] = scenario_name
+    flags["ddagent:deploy"] = install_agent
+    flags["ddagent:version"] = agent_version
+    _deploy_with_config(ctx, flags)
+
+
+def _deploy_with_config(ctx, flags):
+    cfg = config.get_config()
+    flags["ddinfra:aws/defaultKeyPairName"] = cfg["aws_key_pair"]
+    flags["ddinfra:env"] = "aws/sandbox"
+    
+    if "ddagent:deploy" in flags and flags["ddagent:deploy"] :
+        flags["ddagent:apiKey"] = _get_api_key()
+    
+    cmd_args = ["aws-vault", "exec", "sandbox-account-admin", "--", "pulumi", "up"]
+    for (key, value) in flags.items():   
+        if value is not None and value != "":     
+            cmd_args.append("-c")
+            cmd_args.append(shlex.quote(f"{key}={value}"))
+    cmd_args.extend(["-s", _get_stack_name(flags["scenario"])])
+
+    try:
+        # use subprocess instead of context to allow interaction with pulumi up
+        subprocess.check_call(cmd_args)    
+    except Exception as e:
+        raise invoke.Exit(f"Error when running {cmd_args}: {e}")
+
+
+def _get_api_key():    
+    api_key = os.getenv("DD_API_KEY")
+    if api_key is None or len(api_key) != 32:
+        raise invoke.Exit("Invalid API KEY. You must define the environment variable 'DD_API_KEY'. " +
+                            "If you don't want an agent installation add '--no-install-agent'.")
+    return api_key
+
+
+stack_name_prefix = "invoke-"
+
+
+def _get_stack_name(scenario_name):    
```

Comment:
I believe that's too much prefixing. I don't think we really care if the stack was spawned by `invoke` or `pulumi up` or whatever.
However we do care about being able to spawn 2 VMs (so giving name in input).

Reply:
> I don't think we really care if the stack was spawned by invoke or pulumi up or whatever.

I think we care. As long as there is a single environment, calling `inv destroy` uses the same config as `pulumi up` and so the code path is the same (the environment is properly destroyed).

Reply:
Having a prefix does not change this, it's only use to restrict the ability of `inv destroy` to destroy stacks that were created before, which I don't understand why we artificially limit.
However it will fill all object names with `invoke-` everywhere, which, I believe, in some condensed cloud provider UI is just hurting readability.